src/algebra/order/absolute_value.lean

```diff
@@ -94,20 +99,18 @@ variables [nontrivial R]
 (mul_right_inj' $ abv.ne_zero one_ne_zero).1 $
 by rw [← abv.map_mul, mul_one, mul_one]
 
+instance : monoid_with_zero_hom_class (absolute_value R S) R S :=
```

Comment:
Does it have different assumptions than `mul_hom_class`?

Reply:
it seems `map_one` currently requires a `linear_order`. the generalisation linter has a few things to say about this file already, so I'll have a look at this after Yael merges this as to not cause them merge conflicts

Reply:
#16190 is a start. I haven't figured out the best way for `map_one` yet.